Lib/idlelib/textview.py

```diff
@@ -28,52 +29,104 @@ def place(self, **kwargs):
         raise TclError(f'{self.__class__.__name__} does not support "place"')
 
 
-class TextFrame(Frame):
-    "Display text with scrollbar."
+def add_config_callback(config_callback):
+    """Class decorator adding a configuration callback for Tk widgets"""
+    def decorator(cls):
+        class WidgetWithConfigHook(cls):
+            def __setitem__(self, key, value):
+                config_callback({key: value})
+                super().__setitem__(key, value)
 
-    def __init__(self, parent, rawtext, wrap='word'):
+            def configure(self, cnf=None, **kw):
+                if cnf is not None:
+                    config_callback(cnf)
+                if kw:
+                    config_callback(kw)
+                super().configure(cnf=cnf, **kw)
+
+            config = configure
+
+        update_wrapper(WidgetWithConfigHook, cls, updated=[])
+        return WidgetWithConfigHook
+
+    return decorator
+
+
+class ScrollableTextFrame(Frame):
+    """Display text with scrollbar(s)."""
+
+    def __init__(self, master, **kwargs):
         """Create a frame for Textview.
 
-        parent - parent widget for this frame
-        rawtext - text to display
+        master - master widget for this frame
+
+        The Text widget is accessible via the 'text' attribute.
         """
-        super().__init__(parent)
-        self['relief'] = 'sunken'
-        self['height'] = 700
+        super().__init__(master, **kwargs)
 
-        self.text = text = Text(self, wrap=wrap, highlightthickness=0)
-        color_config(text)
-        text.grid(row=0, column=0, sticky=N+S+E+W)
+        @add_config_callback(self._config_callback)
+        class TextWithConfigHook(Text):
+            pass
+        self.text = TextWithConfigHook(self)
```

Comment:
As you understood, I removed this to avoid a name clash with the 'text' input parameters. Since we're changing the input parameter's name, you're welcome to re-introduce the local 'text' variable for the Text widget.